test/local/mysql_tests.js

```diff
@@ -221,6 +221,109 @@ DB.connect(config)
+          .then(function(result) {
+            t.ok(result.count >= 0, 'returned non-negative count')
+          }, function() {
+            t.fail('should have succeeded')
+          })
+        }
+      )
+
+      test(
+        'db.countAccountsCreatedBefore(0)',
+        function (t) {
+          return db.countAccountsCreatedBefore(0)
+          .then(function(result) {
+            // Count should be zero if STRICT_ALL_TABLES and
+            // STRICT_TRANS_TABLES are not set in my.cnf
+            t.plan(1)
+            t.equal(result.count, 0)
+          }, function(err) {
+            // Query should fail if STRICT_ALL_TABLES or
+            // STRICT_TRANS_TABLES are set in my.cnf
+            t.plan(2)
+            t.equal(err.code, 500)
+            t.equal(err.errno, 1264)
+          })
+        }
+      )
+
+      test(
+        'db.countVerifiedAccountsCreatedBefore(Date.now())',
+        function (t) {
+          t.plan(1)
+          return db.countVerifiedAccountsCreatedBefore(Date.now())
+          .then(function(result) {
+            t.ok(result.count >= 0, 'returned non-negative count')
+          }, function() {
+            t.fail('should have succeeded')
+          })
+        }
+      )
+
+      test(
+        'db.countVerifiedAccountsCreatedBefore(0)',
+        function (t) {
+          return db.countVerifiedAccountsCreatedBefore(0)
+          .then(function(result) {
+            // Count should be zero if STRICT_ALL_TABLES and
+            // STRICT_TRANS_TABLES are not set in my.cnf
+            t.plan(1)
+            t.equal(result.count, 0)
+          }, function(err) {
+            // Query should fail if STRICT_ALL_TABLES or
+            // STRICT_TRANS_TABLES are set in my.cnf
+            t.plan(2)
+            t.equal(err.code, 500)
+            t.equal(err.errno, 1264)
+          })
+        }
+      )
+
+      test(
+        'db.countMultiDeviceAccounts()',
+        function (t) {
+          t.plan(1)
+          return db.countMultiDeviceAccounts()
+          .then(function(result) {
+            t.ok(result.count >= 0, 'returned non-negative count')
+          }, function() {
+            t.fail('should have succeeded')
+          })
+        }
+      )
+
+      test(
+        'db.countThreePlusDeviceAccounts()',
+        function (t) {
+          t.plan(1)
+          return db.countThreePlusDeviceAccounts()
+          .then(function(result) {
+            t.ok(result.count >= 0, 'returned non-negative count')
+          }, function() {
+            t.fail('should have succeeded')
+          })
+        }
+      )
+
+      test(
+        'db.countMobileDeviceAccounts()',
+        function (t) {
+          t.plan(1)
+          return db.countMobileDeviceAccounts()
+          .then(function(result) {
+            t.ok(result.count >= 0, 'returned non-negative count')
+          }, function(err) {
+            t.fail('should have succeeded')
+          })
+        }
+      )
```

Comment:
These tests are a great start, but we should also test that they're actually counting the things we care about, by comparing counts after we add/remove sessions.  That will be particularly important if we add more complex filtering logic like last-access-time checking.